server/api/functions/astar.py

```diff
@@ -11,60 +15,57 @@
 """
 
 class Node:
-    def __init__(self, parent=None, position=None):
-        self.parent = parent
-        self.position = position
-        self.cost_to_reach = 0
-        self.total_cost = 0
-        self.heuristic_cost = 0
+    def __init__(self, parent: 'Node' = None, position: Tuple[int, int] = None):
+        self.parent: Node = parent
+        self.position: Tuple[int, int] = position
```

Comment:
don't type hint variables because it causes asymmetry (sorry i thought i mentioned this earlier, if i didn't, let me know and i'll fix it)